### PR TITLE
Add runasuser toggle

### PR DESCRIPTION
**Pull request by gamuniz**
add run_userid to allow uniform UID across task and controller-ee

### CODE REVIEW

**Review by kdelee — APPROVED** [review]

File: roles/installer/templates/deployment.yaml.j2 [hunk 1]
```diff
@@ -183,6 +183,11 @@ spec:
 {% if task_privileged == true %}
           securityContext:
             privileged: true
+{% elif run_userid %}
+          securityContext:
+            runAsUser: {{ run_userid }}
+{% else %}
```
Comment (dsavineau) [inline roles/installer/templates/deployment.yaml.j2]: running a container with a specific user or with the `privileged` flag aren't exclusive (it's not a xor). You can have a privileged container with a non root user so the `elif` statement doesn't right to me.

Comment (dsavineau) [inline roles/installer/templates/deployment.yaml.j2]: What's the purpose of the `else` statement here ? there's nothing to do in this situation so we can drop that statement

**Review by dsavineau — CHANGES_REQUESTED** [review]

File: roles/installer/templates/deployment.yaml.j2 (hunk 1 above)
Comment (gamuniz) [reply roles/installer/templates/deployment.yaml.j2]: I can fix that up, jinja2 isn't my favorite to work out. as for the xor, I didn't want to break existing functionality to fix this for my use-case. If you have a suggestion that works for both. I've never tried setting runasuser and privileged as it seems redundant to use runasuser and privileged: true

**Comment by pabelanger** [comment]
Is this related to https://github.com/ansible/awx-operator/pull/645 ?

**Comment by gamuniz** [comment]
closing in favor of #645

**Pull request closed without merging**